eos_data_distribution/ndn/base.py

```diff
@@ -211,17 +213,17 @@ def registerPrefix(self, prefix=None, flags=None, *args, **kwargs):
         self._prefixes[prefix] = self._registerPrefix(prefix, flags, *args, **kwargs)
         return prefix
 
-    def _registerPrefix(self, prefix, cost=None, controlParameters={},
+    def _registerPrefix(self, prefix, cost=None, controlParameters=None,
                            onInterest=None, onRegisterFailed=None,
                            onRegisterSuccess=None,
                            *args, **kwargs):
         if not cost: cost = self.cost
         if not onInterest: onInterest = self._onInterest
         if not onRegisterFailed: onRegisterFailed = self.onRegisterFailed
         if not onRegisterSuccess: onRegisterSuccess = self.onRegisterSuccess
+        if not controlParameters: controlParameters = ControlParameters()
 
-        if cost: controlParameters['cost'] = int(cost)
-        logger.debug("Register Prefix: %s", controlParameters)
+        if cost: controlParameters.setCost(int(cost))
```

Comment:
If it comes from the command line you should do the cast at the same time you check its validity; and you should probably reject any inputs which are not valid integers.